src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -195,6 +197,15 @@ var ReactCompositeComponentMixin = {
           Component.displayName || Component.name || 'Component'
         );
       }
+
+      propsMutated = !shallowEqual(inst.props, publicProps);
```

Comment:
Because it's more inline with our facebook style, it gives linters an opportunity to catch more errors, and it minimizes the number of variables that a developer needs to keep in their mental working set.